starter_training_script.ipynb

```diff
@@ -0,0 +1,272 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import tensorflow as tf\n",
+    "from datasets import load_dataset\n",
+    "from transformers import AutoTokenizer, create_optimizer, AdamWeightDecay, TFAutoModelForCausalLM\n",
+    "\n",
+    "\n",
+    "#most code is taken from the old huggingface script for language modeling with \n",
+    "DATASET_PATH = \"kpriyanshu256/whisper-transcripts\"\n",
+    "MODEL_CHECKPOINT = \"distilgpt2\"\n",
+    "BLOCK_SIZE = 64\n",
+    "EPOCHS = 1\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 62,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create functions to read in JSON data of transcripts\n",
+    "\n",
+    "def retrieve_dataset(path):\n",
+    "    # given filepath to Hugging Face Dataset, loads the data\n",
+    "    datasets = load_dataset(path)\n",
+    "    return datasets\n",
+    "\n",
+    "\n",
+    "def tokenize_function(dat):\n",
+    "    tokenizer = AutoTokenizer.from_pretrained(MODEL_CHECKPOINT)\n",
+    "    return tokenizer(dat[\"text\"])\n",
+    "\n",
+    "\n",
+    "def group_texts(dat):\n",
```

Comment:
The same in here you have been using global variables inside a function scope. I will create a default value in the parameters of the function to be able to change it.